File: Examples/test-suite/octave/virtual_poly_runme.m
virtual_poly

d = virtual_poly.NDouble(3.5);
i = virtual_poly.NInt(2);

#
# the copy methods return the right polymorphic types
# 
dc = d.copy();
ic = i.copy();

if (d.get() != dc.get())
  error("failed");
endif

if (i.get() != ic.get())
  error("failed");
endif

virtual_poly.incr(ic);

if ((i.get() + 1) != ic.get())
  error("failed");
endif


dr = d.ref_this();
if (d.get() != dr.get())
  error("failed");
endif


#
# 'narrowing' also works
#
ddc = virtual_poly.NDouble_narrow(d.nnumber());
if (d.get() != ddc.get())
  error("failed");
endif

dic = virtual_poly.NInt_narrow(i.nnumber());
if (i.get() != dic.get())
  error("failed");
endif

File: Examples/test-suite/lua/virtual_poly_runme.lua
require("virtual_poly")

local d = virtual_poly.NDouble(3.5)
local i = virtual_poly.NInt(2)

-- polymorphic return type working
local dc = d:copy()
local ic = i:copy()

assert(d:get() == dc:get())
assert(i:get() == ic:get())

-- 'narrowing' working ('nnumber' returns a NNumber)
local ddc = virtual_poly.NDouble.narrow(dc:nnumber())
assert(d:get() == ddc:get())

local dic = virtual_poly.NInt.narrow(ic:nnumber())
assert(i:get() == dic:get())
